packages/firestore/src/platform/browser_lite/connection.ts

```diff
@@ -22,6 +22,6 @@ import { FetchConnection } from './fetch_connection';
 export { newConnectivityMonitor } from '../browser/connection';
 
 /** Initializes the HTTP connection for the REST API. */
-export function newConnection(databaseInfo: DatabaseInfo): Promise<Connection> {
-  return Promise.resolve(new FetchConnection(databaseInfo, fetch.bind(null)));
+export function newConnection(databaseInfo: DatabaseInfo): Connection {
+  return new FetchConnection(databaseInfo, fetch.bind(null));
```

Comment:
Can you explain why we need to bind `fetch` to `null` here?

Reply:
The first argument to any JavaScript function is the "this context", which is this case is not used. It still has to be explicitly set as we are using "bind". If we just pass "bind" as this then calling this function pointer later would use the first argument as the "this" context... and essentially drop it. 
Note that we could also do `fetch.bind(this)`, which would be equivalent to using an array function here.

As an example, if we have a function like this:

```
class Foo {
bar = 'bar';

function foo() {
  console.log(this.bar);
}

function test() {
   this.foo(); // Prints 'bar';
   (() => this.foo())() // Prints 'bar', since arrow functions retain `this` context.
   this.foo().bind({bar:'foo'})(); // Different `this`. Prints 'foo'.
}
```

```